### ISSUE TITLE
unexported-return: does not allow unexpected type aliases for exported types

### ISSUE THREAD

**Issue by tie** [mgechev/revive#1455]
**Describe the bug**

Here is an example code that returns an exported type, but uses type aliases to hide the actual type signature from the documentation.

```go
type unmarshalerFunc = hidden.UnmarshalerFunc

func unmarshal() unmarshalerFunc {
	return func(any) error {
		return nil
	}
}

var unaliasedType func() hidden.UnmarshalerFunc = unmarshal
```

**To Reproduce**
n/a

**Expected behavior**

A function that returns a unexported type alias for an exported type should not trigger the UnexportedReturn rule since it is equivalent to using the unaliased type.

**Logs**
n/a

**Desktop (please complete the following information):**
 - OS: n/a
 - 1.24.5

**Additional context**
n/a

**Comment by denisvmedia** [comment]
Thank you for your report @tie. I personally don't think it's a bug. See my comment in your PR.

**Comment by tie** [comment]
Created https://github.com/mgechev/revive/issues/1457 after discussion in the PR.
